test/lib/PlacesTestUtils.js

```diff
@@ -24,22 +26,69 @@ const PlacesTestUtils = Object.freeze({
    *            [optional] visitDate: visit date in microseconds from the epoch
    *            [optional] referrer: nsIURI of the referrer for this visit
    *          }
-   *
-   * @return {Promise}
-   *            resolves When all visits have been added successfully.
-   *            rejects JavaScript exception.
    */
   addVisits: Task.async(function*(placeInfo) {
-    let promise = new Promise((resolve, reject) => {
-      let places = [];
-      if (placeInfo instanceof Ci.nsIURI) {
-        places.push({uri: placeInfo});
-      } else if (Array.isArray(placeInfo)) {
-        places = places.concat(placeInfo);
-      } else {
-        places.push(placeInfo);
+    let places = [];
+
+    if (placeInfo instanceof Ci.nsIURI) {
+      places.push({uri: placeInfo});
+    } else if (Array.isArray(placeInfo)) {
+      places = places.concat(placeInfo);
+    } else {
+      places.push(placeInfo);
+    }
+
+    // Keep track of the number of rows for final validation
+    function* getPlacesRows() {
```

Comment:
(discussed with @oyiptong already... adding for record keeping)

Just one more note on `yield*`: calling it will run the generator, but, AFAICT, those DB promises will not being handled be handled by any promise-resolving machinery (because you need something like Task.spawn/async to handle passing the resolved value back into the generator).